packages/next/src/server/app-render/app-render.tsx

```diff
@@ -930,7 +985,7 @@ async function renderToHTMLOrFlightImpl(
       // place however it is critical that we only construct the Flight Response inside the SSR
       // render so that directives like preloads are correctly piped through
       const serverStream = ComponentMod.renderToReadableStream(
-        <ReactServerApp tree={tree} ctx={ctx} asNotFound={asNotFound} />,
+        await getRootAppProps(tree, ctx, asNotFound),
```

Comment:
Can we lift this above the call to render? It doesn't change anything but it makes it slightly clearer that we produce the tree outside of the render whereas with this it's a little subtle and it might casually look like the work of this function happens during the render